crates/re_ui/src/list_item2/scope.rs

```diff
@@ -13,18 +12,54 @@ pub struct State {
     // be generalized to some `full_span_scope` mechanism to be used by all full-span widgets beyond
     // `ListItem`.
     pub(crate) background_x_range: egui::Rangef,
-    // TODO(ab): record the use of right action button in all PropertyContent such as to not reserve
-    // right gutter space if none have it.
+
+    /// Left-most X coordinate for the scope.
+    ///
+    /// This is the reference point for tracking column width. This is set by [`list_item_scope`]
+    /// based on `ui.max_rect()`.
+    pub(crate) left_x: f32,
+
+    /// Column width to be used this frame.
+    ///
+    /// The column width has `left_x` as reference, so it includes:
+    /// - All the indentation on the left side of the list item.
+    /// - Any extra indentation added by the list item itself.
+    /// - The list item's collapsing triangle, if any.
+    ///
+    /// The effective left column width for a given [`super::ListItemContent`] implementation can be
+    /// calculated as `left_column_width - (context.rect.left() - left_x)`.
+    pub(crate) left_column_width: Option<f32>,
+
+    /// Maximum desired column width, to be updated this frame.
+    ///
+    /// The semantics are exactly the same as for `left_column_width`.
+    max_desired_left_column_width: f32,
```

Comment:
Does this every shrink, or does it only ever grow?

Reply:
It's reset to -inf each frame by the scope, and only grows during the frame according the needed width.